android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyTestUtil.kt

```diff
@@ -0,0 +1,169 @@
+package org.greatfire.envoy
+
+import android.util.Log
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+
+// MNB: Can you document this? What is it's job?
+
+/*
+This seems to be:
+* managing starting and stopping services
+* managing the active sevice
+* keeping track of statistics
+* reporting back to the caller
+* holding some test related logic
+*/
```

Comment:
it seems like this is just timers and reporting now

Reply:
yeah, I forgot to update that comment

Reply:
I'm not sure it makes sense for that class to be a singleton... it's only created in one place, and if it's not a singleton, we don't have to worry about reset logic... it gets created when a Connection Worker is started, and it's done when the connection is established